docker/package/tezos_baking_wizard.py

```diff
@@ -0,0 +1,489 @@
+            reqs=lambda vs: vs["snapshot"] == "url",
+        ),
+        Step(
+            id="key_import_mode",
+            prompt="How do you want to import the baker key?",
+            help="To register the baker, its secret key needs to be imported to the data "
+            "directory first.\nBy default tezos-baking-<network>.service will use the 'baker' "
+            "alias for the key that will be used for baking and endorsing.\n"
+            "If you want to test baking with a faucet file, make sure you have chosen a test network like "
+            + list(networks.keys())[1],
+            options=pprint_table(key_import_modes),
+            validator=Validator(enum_range_validator(key_import_modes)),
+            reqs=lambda vs: vs["mode"] == "baker",
+        ),
+        Step(
+            id="ledger_url",
+            prompt="Provide the ledger URL for importing the baker secret key stored on a ledger.\n",
+            help="Your ledger URL should look something like <ledger://<mnemonic>/ed25519/0'/1'>",
+            validator=Validator(required_field_validator),
+            reqs=lambda vs: vs["mode"] == "baker" and vs["key_import_mode"] == "ledger",
+        ),
+        Step(
+            id="secret_key",
+            prompt="Provide either the unencrypted or password-encrypted secret key for your address.\n",
+            help="The format is 'unencrypted:edsk...' for the unencrypted key, or 'encrypted:edesk...'"
+            "for the encrypted key.",
+            validator=Validator(required_field_validator),
+            reqs=lambda vs: vs["mode"] == "baker"
+            and vs["key_import_mode"] == "secret-key",
+        ),
+        Step(
+            id="json_filepath",
+            prompt="Provide the path to your downloaded faucet JSON file.\n",
+            help="Download the faucet JSON file from https://faucet.tzalpha.net/.\n"
+            "The file will contain the 'mnemonic' and 'secret' fields.",
+            validator=Validator([required_field_validator, filepath_validator]),
+            reqs=lambda vs: vs["mode"] == "baker" and vs["key_import_mode"] == "json",
+        ),
+    ]
+
+    for step in steps:
+        if not step.reqs(values):
+            continue
+
+        validated = False
+        while not validated:
+            print(step.prompt)
+            if step.options is not None:
+                print(step.options)
+            answer = input("> ").strip()
+
+            if answer.lower() in ["quit", "exit"]:
+                print("Exiting Tezos Baking Wizard")
+                return None
+            elif answer.lower() in ["help", "?"]:
+                print(step.help)
+            else:
+                if not answer and step.default is not None:
+                    answer = step.default
+
+                try:
+                    answer = step.validator.validate(answer)
+                except ValueError as e:
+                    print(color("Validation error: " + str(e), "\x1b[1;31m"))
```

Comment:
likewise I think that most of this "handling validation" logic could be moved to a `validate` method of `Step`.

Reply:
That's how it was initially, see https://github.com/serokell/tezos-packaging/pull/206#pullrequestreview-667534646

Reply:
So, you know, tell me which one of you you think knows better :smile:

Reply:
:joy:

Reply:
I don't have a really strong opinion about this, but IMO it's better when validator logic is hidden inside the validator, i.e. we don't know whether the validator accepts only a fixed list of options or only filepaths or something different

Reply:
Oh, my comment was unclear, I actually agree with that.

My suggestion was only about moving the handling this `try` and the `validated` variable inside the `step`, which isn't related to the validator itself, but to the validation process in general.

Now that I looks at it again, I suppose one way to do that would be to start without a:
```python
validated = False
while not validated
```
and instead use a:
```python
answer = None
while answer is None
```
and instead of the `try` block call:
```python
answer = step.validate(answer)
```
which actually contains this `try` and returns `None` on failure.

It's not much better TBH, it may even be worse to follow, I don't have a strong opinion either and I'll let you decide :slightly_smiling_face:

Reply:
I tried it out, and I believe it's actually much worse to follow. With validation logic and the _validator_ logic separate, it now comes to calling `step.validate` which in turn calls `step.validator.validate`, and also `values[step.id]` is constantly overwritten until the condition in the `while` clause stops to be met (and the condition is not as immediately understandable, I believe). I don't think we would gain much from this separation, but it seems we do lose some at-a-glance readability :/ So I'll leave it as is for now